1_assignment/unflatten.js

```diff
@@ -0,0 +1,42 @@
+// Sample Complex flattened JSON object.
+var data = {"flatJSON": false, "i.am.not.so.flat": true, "i.am.not.so.unflat": false, "i.am.a": "tree", "dates.0.day": 1, "dates.1.day": 8947};
+
+function unflatten(flatObject){
+  var temp_str;
+  var unflatObject = {};
+  for (obj in flatObject){  
+    var temp_object = {};
+    temp_str = String(obj);
+    var splitted_string = temp_str.split(".");
+    // call to form_object function with the copy of reference for unflatObject.
+    form_object(unflatObject , splitted_string , flatObject[obj]);
+  }   
+  return unflatObject;
+}
+
+function form_object(unflatObject , splitted_string , value){
```

Comment:
Space before starting the block. viz, `form_object(unflatObject, splitted_string, value) {`